### PR TITLE
Kinect WSJ Recipe

### PR DESCRIPTION
**Pull request by sunits**
1. Contains recipe to train Deep Clustering on Kinect WSJ

TODO:
Multichannel  Deep Clustering

### CODE REVIEW

File: egs/kinect-wsj/DeepClustering/local/preprocess_wsj0mix.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import argparse
```
Comment (mpariente) [inline egs/kinect-wsj/DeepClustering/local/preprocess_wsj0mix.py:1]: Let's rename this file to `preprocess_kinectwsj0mix` or something like that.

File: egs/kinect-wsj/DeepClustering/local/preprocess_wsj0mix.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    speaker_list = ['mix'] + [f"s{n+1}" for n in range(inp_args.n_src)]
+    for data_type in ['tr', 'cv', 'tt']:
+        for spk in speaker_list:
+            preprocess_one_dir(os.path.join(inp_args.in_dir, data_type, spk),
+                               os.path.join(inp_args.out_dir, data_type),
+                               spk)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("WHAM data preprocessing")
```
Comment (mpariente) [inline egs/kinect-wsj/DeepClustering/local/preprocess_wsj0mix.py:36]: And have the right help message (I'll fix that in the wsj0mix recipe as well)

File: egs/kinect-wsj/DeepClustering/run.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+sr_string=$(($sample_rate/1000))
+suffix=${n_src}speakers/wav${sr_string}k/$mode
+dumpdir=data/$suffix  # directory to put generated json file
+
+train_dir=$dumpdir/tr
+valid_dir=$dumpdir/cv
+test_dir=$dumpdir/tt
+
+if [[ $stage -le  0 ]]; then
+    echo "Create wsj0-mix files"
```
Comment (mpariente) [inline egs/kinect-wsj/DeepClustering/run.sh:63]: `echo "Create wsj0-mix files and start again from stage 1" || exit 1`

File: egs/kinect-wsj/DeepClustering/train.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    trainer.fit(system)
+
+    with open(os.path.join(exp_dir, "best_k_models.json"), "w") as f:
+        json.dump(checkpoint.best_k_models, f, indent=0)
+    # Save last model for convenience
+    torch.save(system.model.state_dict(),
+               os.path.join(exp_dir, 'checkpoints/final.pth'))
+
+
+    # TODO:Should ideally be inherited from wsj0-mix
```
Comment (mpariente) [inline egs/kinect-wsj/DeepClustering/train.py:84]: I'd love to reduce the copy pasting from recipes to recipes but apart from adding this things into the source code, I didn't find a good solution yet. Any good idea welcome !

File: asteroid/data/__init__.py [hunk 5]
```diff
@@ -4,3 +4,4 @@
 from .librimix_dataset import LibriMix
 from .wsj0_mix import Wsj0mixDataset
 from .musdb18_dataset import MUSDB18Dataset
+from .kinect_wsj import KinectWsjMixDataset
```
Comment (mpariente) [inline asteroid/data/__init__.py:7]: New line at the end of the file please.

**Comment by mpariente** [comment]
Tests are failing because Zenodo's sandbox is down, no link to your PR.

File: egs/kinect-wsj/DeepClustering/train.py (hunk 4 above)
Comment (sunits) [reply egs/kinect-wsj/DeepClustering/train.py:84]: Is Deep clustering the only recipe which requires a custom System class?

Comment (mpariente) [reply egs/kinect-wsj/DeepClustering/train.py:84]: No, also in DNS challenge and the two step recipe

Comment (sunits) [reply egs/kinect-wsj/DeepClustering/train.py:84]: DC models can probably go into blocks.
System callbacks can handle common_steps I think. Will have to analyse a bit more to see if it gets complicated.

File: egs/kinect-wsj/DeepClustering/run.sh [hunk 6]
```diff
@@ -60,7 +60,7 @@ valid_dir=$dumpdir/cv
 test_dir=$dumpdir/tt
 
 if [[ $stage -le  0 ]]; then
-    echo "Create wsj0-mix files"
+    echo "Create wsj0-mix files and start again from stage 1" || exit 1
```
Comment (mpariente) [inline egs/kinect-wsj/DeepClustering/run.sh:63]:
```suggestion
     echo "Create wsj0-mix files and start again from stage 1" ; exit 1
```

**Comment by mpariente** [comment]
Thanks that's perfect for now !
"maskter": muscle memory spotted :wink:

**Pull request merged**